src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionUsage.java

```diff
@@ -130,10 +131,26 @@ public final boolean getHasNonDevUseExtension() {
     return getProxies().stream().anyMatch(p -> !p.isDevDependency());
   }
 
+  @GenerateTypeAdapter
+  public record RepoOverride(
+      // The apparent name of the overriding repo in the root module.
+      String overridingRepoName,
+      // Whether this override should apply to an existing repo.
+      boolean mustExist,
+      Location location) {}
+
+  /**
+   * Maps repo names local to the extension to repo they are overridden by.
+   *
+   * <p>This is only non-empty for root module usages and repos that override other repos are not
+   * themselves overridden.
+   */
+  public abstract ImmutableMap<String, RepoOverride> getRepoOverrides();
+
   public abstract Builder toBuilder();
 
   public static Builder builder() {
-    return new AutoValue_ModuleExtensionUsage.Builder();
+    return new AutoValue_ModuleExtensionUsage.Builder().setRepoOverrides(ImmutableMap.of());
```

Comment:
is this default value necessary? seems like we should make sure it's set in prod code, in exchange for a bit more verbose test code.